gcore/gdalpamdataset.cpp

```diff
@@ -66,7 +66,9 @@ CPL_CVSID("$Id$")
  * PAM support can be enabled (resp. disabled) in GDAL by setting the
  * GDAL_PAM_ENABLED configuration option (via CPLSetConfigOption(), or the
  * environment) to the value of YES (resp. NO). Note: The default value is
- * build dependent and defaults to YES in Windows and Unix builds.
+ * build dependent and defaults to YES in Windows and Unix builds. Warning:
+ * Setting this option to OFF may have unwanted side-effects on drivers
```

Comment:
```suggestion
 * For GDAL < 3.5, setting this option to OFF may have unwanted side-effects on drivers
```
(cf fix of https://github.com/OSGeo/gdal/pull/5511)

Reply:
Why not to say that the side effect is "CPLError() to be emitted"?

Reply:
> 

because the side effect is also (for example in my case) that the band's colorinterpretation was not being set

Reply:
I think that anybody who disables PAM has considered that it does not contain any useful data so I guess that colorinterpretation is stored into DIMAP XML metadata and it was skipped because aux.xml was not found.

Reply:
> so I guess that colorinterpretation is stored into DIMAP XML metadata and it was skipped because aux.xml was not found.

that's more complicated than that. A number of drivers have GDALMyDriverNameDataset / GDALMyDriverNameRasterBand classes that extend the base GDALPamDataset / GDALPamRasterBand classes, in case loading / saving .aux.xml is needed (e.g. the generic function of saving statistics, as used by QGIS, in .aux.xml depend on the drivers extending those classes). But the drivers may just do very basic things like someBand->SetColorInterpretation(GCI_RedBand) during dataset opening, that do not require any serialization capabilities at all. Except that before the fixed queued in #5511, that wouldn't work when PAM is disabled.

Reply:
No, that's not exactly the case. The colorinterpretation is written in the dimap xml, but set programmatically in a PamDataset by the dimap driver at GDALOpen() time (https://github.com/OSGeo/gdal/blob/af52706a15cdc145fc4b740a5ee29229cca2412b/frmts/dimap/dimapdataset.cpp#L1388). This causes an error when GDAL_PAM_ENABLED is set to OFF, whether or not the aux.xml file is present or not.